### PR TITLE
Some smaller contributions collected in one PR

### PR DESCRIPTION
**Pull request by bs-jokri**
- Replace findbugs.annotations in ektorp lib with Apache Licensed reimplementation
- Allow FOSSology keys to be placed at `/etc/sw360` 
   This also introduces a new function CommonUtils.loadResource which generalizes the placement of resources at /etc/sw360
- if `Name` of an SPDX license in imported spdx file is not filled, use its `LicenseId` as license name
- setting project default values for new projects
   - state = active
   - visibility = group+moderators
   - projectType = customer
   - moderators = []
   - contributors = []
   - clearing team = unknown
- set Liferay SessionError instead of SessionMessage for license duplicate
- generalize location of thrift binary
   since thrift gets installed into $PATH, there is no need to add the fully qualified path to the binary.

   Even the Ubuntu package installs it into another folder /usr/bin/ instead of /usr/local/bin/ and there are other location where the binary can be found.

### CODE REVIEW

**Comment by mcjaeger** [comment]
I think we have conflicting default values here: for us, it is product and visibility public. How about having the default type being possible to set in sw360*.properties?

**Comment by bs-jokri** [comment]
Good idea. @maximilianhuber what do you think?

**Comment by maximilianhuber** [comment]
Yes that is a good idea. And we have to rebase this PR.

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java [hunk 1]
```diff
@@ -505,6 +508,35 @@ public static Properties loadProperties(Class<?> clazz, String propertiesFilePat
         return props;
     }
 
+    public static Optional<byte[]> loadResource(Class<?> clazz, String resourceFilePath) {
+        return loadResource(clazz, resourceFilePath, true);
+    }
+
+    public static Optional<byte[]> loadResource(Class<?> clazz, String resourceFilePath, boolean useSystemRecourses) {
```
Comment (alex-evo) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java]: Like this.
btw, `useSystemRecourses` is spelled incorrectly

**Comment by maximilianhuber** [comment]
@mcjaeger The way this PR currently defines the default visibility of projects does not allow configuration via a properties file. It is [hardcoded in the thrift file](https://github.com/sw360/sw360portal/pull/309/commits/7fe183dc9760b32a00387a219e0162c23614ceb7#diff-58a35ca111021fb800adea75699e6dcaR78).

**Comment by mcjaeger** [comment]
I think I messed something up. It seems like it is Ok for now.

**Comment by maximilianhuber** [comment]
There are new test failures. I will have a look and correct the rebase-failure.

**Comment by mcjaeger** [comment]
@bs-jokri @maximilianhuber could you pls more helpful with some comments in terms of how to test this? so, "use SPDX-LicenseId as license name" is about the SPDXParser.java ?

**Comment by mcjaeger** [comment]
@bs-jokri @maximilianhuber in case, could you add an spdx example file, it seems like the SPDXrdfExample.20 from the SPDX Java tools is not showing this effect properly - maybe, but I am not sure, what to actually look for. For example, for this file I am missing LGPL, but I am not sure, it is correct, if it would be listed?

**Comment by maximilianhuber** [comment]
@mcjaeger I have updated the line related to your question. I hope that it is clearer now.

**Comment by maximilianhuber** [comment]
@mcjaeger the appearance of LGPL is not affected by this PR. I think you have discovered an independent bug which has to be investigated.

**Comment by mcjaeger** [comment]
how about adding test file. So I observed
* only spdx:licenseId -> this is used
* both spdx:licenseId and spdx:name -> name is used
* only spdx:name -> white page returned. and nullptr exception on server. @maximilianhuber have you observed this too?

**Pull request merged**